File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/fraction/shared/designation.service.ts
import { Injectable, Directive } from '@angular/core';
import { DesignationEndPoint } from "./Designation.endpoint";
import { DesignationModel } from './Designation.model';

@Injectable()
export class DesignationService {

    constructor(public DesignationEndpoint: DesignationEndPoint) {

    }

    public GetDesignationList() {
        return this.DesignationEndpoint.GetDesignationList()
            .map(res => { return res });
    }

    public AddDesignation(CurrentDesignation: DesignationModel) {
        return this.DesignationEndpoint.AddDesignation(CurrentDesignation)
            .map(res => { return res });
    }

    public UpdateDesignation(id: number,CurrentDesignation: DesignationModel) {
        return this.DesignationEndpoint.UpdateDesignation(id, CurrentDesignation)
            .map(res => { return res});
    }

    public GetDesignation(id: number) {
        return this.DesignationEndpoint.GetDesignation(id)
            .map(res => { return res });
    }
}

File: Code/Websites/DanpheEMR/Services/Fraction/DesignationService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DanpheEMR.DalLayer;
using DanpheEMR.Core.Configuration;
using Microsoft.Extensions.Options;
using DanpheEMR.ServerModel;

namespace DanpheEMR.Services
{
    public class DesignationService : IDesignationService
    {
        public FractionDbContext db;
        private readonly string connString = null;

        public DesignationService(IOptions<MyConfiguration> _config)
        {
            connString = _config.Value.Connectionstring;
            db = new FractionDbContext(connString);
        }

        public List<DesignationModel> ListDesignation()
        {
            //todo:add server side pagination
            var query = db.Designation.ToList();
            return query;
        }

        public DesignationModel AddDesignation(DesignationModel model)
        {
            model.CreatedOn = DateTime.Now;
            db.Designation.Add(model);
            db.SaveChanges();
            return model;
        }

        public DesignationModel UpdateDesignation(DesignationModel model)
        {
            db.Entry(model).State = System.Data.Entity.EntityState.Modified;
            db.SaveChanges();
            return model;
        }

        public DesignationModel GetDesignation(int id)
        {
            var result = db.Designation.Where(x => x.DesignationId == id).FirstOrDefault();      
            return result;
        }       
    }
}
